cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/DefaultHBaseDDLExecutor.java

```diff
@@ -45,9 +45,9 @@
 public abstract class DefaultHBaseDDLExecutor implements HBaseDDLExecutor {
   public static final Logger LOG = LoggerFactory.getLogger(DefaultHBaseDDLExecutor.class);
   public static final long MAX_CREATE_TABLE_WAIT = 5000L;    // Maximum wait of 5 seconds for table creation.
-  private final Configuration hConf;
+  private Configuration hConf;
 
-  public DefaultHBaseDDLExecutor(Configuration hConf) {
+  void setHConf(Configuration hConf) {
```

Comment:
one possibility is to pass in some HBaseDDLContext interface instead of Configuration, and the context has a `String getProperty(String key)` method, and a `<T> T getHadoopConfiguration()` method similar to what is done in MapReduceContext.getHadoopJob(). The implementation would just call get() on a CConfiguration for getProperty() and would just return the hConf for getHadoopConfiguration().